docs/src/token.md

```diff
@@ -476,3 +476,139 @@ have a balance of zero to be closed.
+* SPL Associated Token Account program - this program defines the convention and the
+  provides the mechanism for mapping the user's wallet address to the associated
+  token accounts they hold.
+
+### Associated Token Account
+The associated token account convention allows all tokens to have the same
+destination address, allowing the user share their main wallet address to
+receive any SPL token.
+
+The following Rust function can be used to derive the address of a user's
+associated token account for a given SPL Token mint:
+```rust
+/// Finds the default token address, and its corresponding bump_seed, for a given wallet
+/// address and SPL Token mint
+pub fn get_associated_token_address(
+    primary_account_address: &Pubkey,
+    spl_token_mint_address: &Pubkey,
+) -> Pubkey {
+    Pubkey::find_program_address(
+        &[
+            &primary_account_address.to_bytes(),
+            &spl_token::id().to_bytes(),
+            &spl_token_mint_address.to_bytes(),
+        ],
+        &spl_associated_token_account::id()
+    ).0
+}
+```
+
+
+Javascript equivalent:
+```
+import {PublicKey, PublicKeyNonce} from '@solana/web3.js';
+
+async function findDefaultTokenAddress(
+    walletAddress: Pubkey,
+    tokenMintAddress: Pubkey
+): Promise<PublicKeyNonce> {
+    return PublicKey.findProgramAddress(
+        [
+            walletAddress.toBuffer(),
+            TOKEN_PROGRAM_ID.toBuffer(),
+            tokenMintAddress.toBuffer(),
+        ],
+        SPL_ASSOCIATED_TOKEN_ACCOUNT_PROGRAM_ID
+    );
+}
+```
+
+### Ancillary Token Accounts
+At any time ownership of an existing SPL Token account may be assigned to the
+user.  One way to accomplish this is with the
+`spl-token authorize <TOKEN_ADDRESS> owner <USER_ADDRESS>` command.
+
+By convention it is suggested that wallets roll up the balances from all token
+accounts of the same mint into a single balance for the user to shield the user
+from this complexity.
+
+### How to fetch and display token holdings
+The [getTokenAccountsByOwner](https://docs.solana.com/apps/jsonrpc-api#gettokenaccountsbyowner)
+JSON RPC method can be used to fetch all token accounts for a wallet address.
+
+### Default Token Account Funding
+Before the user can receive tokens their associated token account must be created
+on-chain, requiring a small amount of SOL to mark the account as rent-exempt.
+
+There's no restriction on who can create a user's associated token account.  It
+could either be created by the wallet on behalf of the user or funded by a 3rd
+party through an airdrop campaign.
+
+Ultimately the
+`spl_associated_token_account::create_associated_token_account()` instruction
+just needs to be executed by some party.
+
+### Associated Token Account Ownership
+The wallet should never use `TokenInstruction::SetAuthority` to set the
+`AccountOwner` authority of the associated token account to another address.
+
+#### Sample "Add Token" workflow
```

Comment:
Add a note explaining why this is needed..  (user wants to receive tokens from other people)